sdk/core/az_core/src/az_http_policy.c

```diff
@@ -70,11 +60,16 @@ AZ_NODISCARD az_result az_http_pipeline_policy_credential(
     _az_http_request* p_request,
     az_http_response* p_response)
 {
-  if (p_data != AZ_CREDENTIAL_ANONYMOUS)
```

Comment:
Leave this as a PRECONDITION

Reply:
As I said below, AZ_CREDENTIAL_ANONYMOUS  is defined as null, so we shouldn't check `NULL->policy_credential_apply == NULL` without checking for NULL first